### ISSUE TITLE
Strange (incorrect) depth image on Waymo dataset

### ISSUE THREAD

**Issue by Crescent-Saturn** [georghess/neurad-studio#28]
Hi there, 

I've been working with the Waymo dataset using the `neurad` method of configuration settings, following the guidelines outlined in the associated paper. However, I'm encountering unexpected results that need further investigation. 
Here is the comparison with `Pandaset` seq `011`:

![image](https://github.com/georghess/neurad-studio/assets/18294546/4a2c4f5e-4a87-42c8-bab8-a84bbcd7a6df)

**Strange Depth Image:** The depth image generated appears to be incorrect. While it correctly depicts the sky as distant, it also portrays the ground as "far," which doesn't align with expectations.

**Blurry Right Sides in Rendered RGB Image:** In both scenes, the right sides of the rendered RGB images appear very blurry.
![image](https://github.com/georghess/neurad-studio/assets/18294546/846230d7-e300-4f2e-b418-676de3d9437c)
![image](https://github.com/georghess/neurad-studio/assets/18294546/50ec39f4-9081-40b5-80ad-c3f129ea7ba6)

Here is the training loss profile:

![image](https://github.com/georghess/neurad-studio/assets/18294546/3dbf32c7-8290-478e-b290-599e84cc2255)

I've double-checked my implementation against the paper's specifications, but I'm still unable to pinpoint the source of these discrepancies.

Any hints or ideas you could provide to help diagnose and resolve these issues would be appreciated.

Thanks for your attention to this matter.

**Comment by Crescent-Saturn** [comment]
Hi @georghess,

Thank you for the kind reply.

This issue has been solved recently as we have found some error in lidar points when export/convert Waymo into NeuRAD dataparser.

I close this as solved. 
Thank you again for your great work!

**Comment by georghess** [comment]
Great! Feel free to open a PR with your parser btw  :)

**Comment by MartinEthier** [comment]
@Crescent-Saturn Did you happen to account for Waymo's rolling shutter? As far as I know, the Waymo dataset uses a column-by-column rolling shutter, but it seems like the rolling shutter compensation in the neurad code compensates for row-by-row rolling shutter, so it would require some minor modifications.

**Comment by Crescent-Saturn** [comment]
> @Crescent-Saturn Did you happen to account for Waymo's rolling shutter? As far as I know, the Waymo dataset uses a column-by-column rolling shutter, but it seems like the rolling shutter compensation in the neurad code compensates for row-by-row rolling shutter, so it would require some minor modifications.

@MartinEthier Thank you for this kind remind. Yeah, it was one of the reason that caused this poor performance. In addition to column-by-column rolling shutter in Waymo, it shows that the direction depends on the sequence. As far as we found, in some sequences the rolling shutter direction is `LEFT_2_RIGHT`, in others it is `RIGHT_2_LEFT`. Thus, we made the corresponding modifications on considering this.

**Comment by MartinEthier** [comment]
Yeah I also noticed that the direction can change based on the camera. Would you mind providing your modification to the rolling shutter compensation code? I'm also working with the Waymo dataset and it would save me some trouble lol

**Comment by Crescent-Saturn** [comment]
From here:
https://github.com/georghess/neurad-studio/blob/445f41d742a01f33437b3367f5ca9b88a86e3e3c/nerfstudio/cameras/cameras.py#L922-L929

we modified `L924-L926` into for column-by-column:
```python
offsets = self.metadata["rolling_shutter_offsets"][cam_idx]
duration = offsets.diff()

width, cols = self.width[cam_idx], coords[..., 1:2]
time_offsets = cols / width * duration + offsets[..., 0:1]
```

hope this helps.
